scripts/src/public/services/orders.ts

```diff
@@ -162,9 +162,10 @@ export async function createExternalOrder(jwt: string, user: User, logger: Logge
 	const payload = await validateExternalOrderJWT(jwt, user.appUserId, logger);
 
 	let order = await db.Order.findOne({ userId: user.id, offerId: payload.offer.id });
+	const completed_orders = await db.Order.getCompletedOrder( user.id, payload.offer.id);
```

Comment:
you can move the completed orders query after the first if, as it's relevant only after that and not in all cases